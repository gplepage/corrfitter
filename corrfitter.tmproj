<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
<dict>
	<key>currentDocument</key>
	<string>setup.py</string>
	<key>documents</key>
	<array>
		<dict>
			<key>filename</key>
			<string>corrfitter.py</string>
			<key>lastUsed</key>
			<date>2012-05-28T19:11:53Z</date>
		</dict>
		<dict>
			<key>filename</key>
			<string>dataset.py</string>
			<key>lastUsed</key>
			<date>2012-05-28T19:12:58Z</date>
		</dict>
		<dict>
			<key>children</key>
			<array>
				<dict>
					<key>filename</key>
					<string>CHANGES</string>
					<key>lastUsed</key>
					<date>2012-05-28T19:39:32Z</date>
				</dict>
				<dict>
					<key>filename</key>
					<string>setup.py</string>
					<key>lastUsed</key>
					<date>2012-05-29T00:18:09Z</date>
					<key>selected</key>
					<true/>
				</dict>
				<dict>
					<key>filename</key>
					<string>MANIFEST.in</string>
					<key>lastUsed</key>
					<date>2012-05-28T19:12:02Z</date>
				</dict>
				<dict>
					<key>filename</key>
					<string>README</string>
					<key>lastUsed</key>
					<date>2012-05-28T19:12:24Z</date>
				</dict>
				<dict>
					<key>filename</key>
					<string>INSTALLATION</string>
					<key>lastUsed</key>
					<date>2012-05-28T19:12:28Z</date>
				</dict>
				<dict>
					<key>filename</key>
					<string>COPYING</string>
					<key>lastUsed</key>
					<date>2012-05-21T02:50:23Z</date>
				</dict>
				<dict>
					<key>filename</key>
					<string>makefile</string>
					<key>lastUsed</key>
					<date>2012-05-21T02:53:49Z</date>
				</dict>
			</array>
			<key>expanded</key>
			<true/>
			<key>name</key>
			<string>build files</string>
		</dict>
		<dict>
			<key>expanded</key>
			<true/>
			<key>name</key>
			<string>tests</string>
			<key>regexFolderFilter</key>
			<string>!.*/(\.[^/]*|CVS|_darcs|_MTN|\{arch\}|blib|.*~\.nib|.*\.(framework|app|pbproj|pbxproj|xcode(proj)?|bundle))$</string>
			<key>sourceDirectory</key>
			<string>tests</string>
		</dict>
		<dict>
			<key>name</key>
			<string>doc</string>
			<key>regexFolderFilter</key>
			<string>!.*/(\.[^/]*|CVS|_darcs|_MTN|\{arch\}|blib|.*~\.nib|.*\.(framework|app|pbproj|pbxproj|xcode(proj)?|bundle))$</string>
			<key>sourceDirectory</key>
			<string>doc</string>
		</dict>
	</array>
	<key>fileHierarchyDrawerWidth</key>
	<integer>200</integer>
	<key>metaData</key>
	<dict>
		<key>CHANGES</key>
		<dict>
			<key>caret</key>
			<dict>
				<key>column</key>
				<integer>28</integer>
				<key>line</key>
				<integer>25</integer>
			</dict>
			<key>firstVisibleColumn</key>
			<integer>0</integer>
			<key>firstVisibleLine</key>
			<integer>0</integer>
		</dict>
		<key>COPYING</key>
		<dict>
			<key>caret</key>
			<dict>
				<key>column</key>
				<integer>46</integer>
				<key>line</key>
				<integer>14</integer>
			</dict>
			<key>firstVisibleColumn</key>
			<integer>0</integer>
			<key>firstVisibleLine</key>
			<integer>0</integer>
		</dict>
		<key>INSTALLATION</key>
		<dict>
			<key>caret</key>
			<dict>
				<key>column</key>
				<integer>25</integer>
				<key>line</key>
				<integer>22</integer>
			</dict>
			<key>columnSelection</key>
			<false/>
			<key>firstVisibleColumn</key>
			<integer>0</integer>
			<key>firstVisibleLine</key>
			<integer>0</integer>
			<key>selectFrom</key>
			<dict>
				<key>column</key>
				<integer>0</integer>
				<key>line</key>
				<integer>22</integer>
			</dict>
			<key>selectTo</key>
			<dict>
				<key>column</key>
				<integer>0</integer>
				<key>line</key>
				<integer>24</integer>
			</dict>
		</dict>
		<key>MANIFEST.in</key>
		<dict>
			<key>caret</key>
			<dict>
				<key>column</key>
				<integer>67</integer>
				<key>line</key>
				<integer>0</integer>
			</dict>
			<key>firstVisibleColumn</key>
			<integer>0</integer>
			<key>firstVisibleLine</key>
			<integer>0</integer>
		</dict>
		<key>README</key>
		<dict>
			<key>caret</key>
			<dict>
				<key>column</key>
				<integer>18</integer>
				<key>line</key>
				<integer>15</integer>
			</dict>
			<key>firstVisibleColumn</key>
			<integer>0</integer>
			<key>firstVisibleLine</key>
			<integer>0</integer>
		</dict>
		<key>corrfitter.py</key>
		<dict>
			<key>caret</key>
			<dict>
				<key>column</key>
				<integer>17</integer>
				<key>line</key>
				<integer>0</integer>
			</dict>
			<key>firstVisibleColumn</key>
			<integer>0</integer>
			<key>firstVisibleLine</key>
			<integer>0</integer>
		</dict>
		<key>dataset.py</key>
		<dict>
			<key>caret</key>
			<dict>
				<key>column</key>
				<integer>15</integer>
				<key>line</key>
				<integer>4</integer>
			</dict>
			<key>firstVisibleColumn</key>
			<integer>0</integer>
			<key>firstVisibleLine</key>
			<integer>0</integer>
		</dict>
		<key>doc/source/Makefile</key>
		<dict>
			<key>caret</key>
			<dict>
				<key>column</key>
				<integer>0</integer>
				<key>line</key>
				<integer>0</integer>
			</dict>
			<key>firstVisibleColumn</key>
			<integer>0</integer>
			<key>firstVisibleLine</key>
			<integer>0</integer>
		</dict>
		<key>doc/source/conf.py</key>
		<dict>
			<key>caret</key>
			<dict>
				<key>column</key>
				<integer>0</integer>
				<key>line</key>
				<integer>0</integer>
			</dict>
			<key>firstVisibleColumn</key>
			<integer>0</integer>
			<key>firstVisibleLine</key>
			<integer>0</integer>
		</dict>
		<key>doc/source/corrfitter.rst</key>
		<dict>
			<key>caret</key>
			<dict>
				<key>column</key>
				<integer>71</integer>
				<key>line</key>
				<integer>28</integer>
			</dict>
			<key>columnSelection</key>
			<false/>
			<key>firstVisibleColumn</key>
			<integer>0</integer>
			<key>firstVisibleLine</key>
			<integer>0</integer>
			<key>selectFrom</key>
			<dict>
				<key>column</key>
				<integer>64</integer>
				<key>line</key>
				<integer>28</integer>
			</dict>
			<key>selectTo</key>
			<dict>
				<key>column</key>
				<integer>71</integer>
				<key>line</key>
				<integer>28</integer>
			</dict>
		</dict>
		<key>doc/source/index.rst</key>
		<dict>
			<key>caret</key>
			<dict>
				<key>column</key>
				<integer>0</integer>
				<key>line</key>
				<integer>14</integer>
			</dict>
			<key>firstVisibleColumn</key>
			<integer>0</integer>
			<key>firstVisibleLine</key>
			<integer>0</integer>
		</dict>
		<key>makefile</key>
		<dict>
			<key>caret</key>
			<dict>
				<key>column</key>
				<integer>3</integer>
				<key>line</key>
				<integer>5</integer>
			</dict>
			<key>firstVisibleColumn</key>
			<integer>0</integer>
			<key>firstVisibleLine</key>
			<integer>0</integer>
		</dict>
		<key>setup.py</key>
		<dict>
			<key>caret</key>
			<dict>
				<key>column</key>
				<integer>30</integer>
				<key>line</key>
				<integer>9</integer>
			</dict>
			<key>firstVisibleColumn</key>
			<integer>0</integer>
			<key>firstVisibleLine</key>
			<integer>0</integer>
		</dict>
		<key>tests/README</key>
		<dict>
			<key>caret</key>
			<dict>
				<key>column</key>
				<integer>21</integer>
				<key>line</key>
				<integer>12</integer>
			</dict>
			<key>firstVisibleColumn</key>
			<integer>0</integer>
			<key>firstVisibleLine</key>
			<integer>0</integer>
		</dict>
		<key>tests/coarse_3pt_1link_Ds_and_etas_kinCth1.10.576.bint8</key>
		<dict>
			<key>caret</key>
			<dict>
				<key>column</key>
				<integer>0</integer>
				<key>line</key>
				<integer>0</integer>
			</dict>
			<key>firstVisibleColumn</key>
			<integer>0</integer>
			<key>firstVisibleLine</key>
			<integer>0</integer>
		</dict>
		<key>tests/coarse_3pt_etas_Ds.bint8</key>
		<dict>
			<key>caret</key>
			<dict>
				<key>column</key>
				<integer>0</integer>
				<key>line</key>
				<integer>0</integer>
			</dict>
			<key>firstVisibleColumn</key>
			<integer>0</integer>
			<key>firstVisibleLine</key>
			<integer>88</integer>
		</dict>
		<key>tests/makefile</key>
		<dict>
			<key>caret</key>
			<dict>
				<key>column</key>
				<integer>0</integer>
				<key>line</key>
				<integer>19</integer>
			</dict>
			<key>firstVisibleColumn</key>
			<integer>0</integer>
			<key>firstVisibleLine</key>
			<integer>0</integer>
		</dict>
		<key>tests/test-1.py</key>
		<dict>
			<key>caret</key>
			<dict>
				<key>column</key>
				<integer>26</integer>
				<key>line</key>
				<integer>97</integer>
			</dict>
			<key>columnSelection</key>
			<false/>
			<key>firstVisibleColumn</key>
			<integer>0</integer>
			<key>firstVisibleLine</key>
			<integer>52</integer>
			<key>selectFrom</key>
			<dict>
				<key>column</key>
				<integer>22</integer>
				<key>line</key>
				<integer>97</integer>
			</dict>
			<key>selectTo</key>
			<dict>
				<key>column</key>
				<integer>26</integer>
				<key>line</key>
				<integer>97</integer>
			</dict>
		</dict>
		<key>tests/test-1b.out</key>
		<dict>
			<key>caret</key>
			<dict>
				<key>column</key>
				<integer>0</integer>
				<key>line</key>
				<integer>0</integer>
			</dict>
			<key>firstVisibleColumn</key>
			<integer>0</integer>
			<key>firstVisibleLine</key>
			<integer>0</integer>
		</dict>
		<key>tests/test-1b.py</key>
		<dict>
			<key>caret</key>
			<dict>
				<key>column</key>
				<integer>26</integer>
				<key>line</key>
				<integer>120</integer>
			</dict>
			<key>columnSelection</key>
			<false/>
			<key>firstVisibleColumn</key>
			<integer>0</integer>
			<key>firstVisibleLine</key>
			<integer>75</integer>
			<key>selectFrom</key>
			<dict>
				<key>column</key>
				<integer>22</integer>
				<key>line</key>
				<integer>120</integer>
			</dict>
			<key>selectTo</key>
			<dict>
				<key>column</key>
				<integer>26</integer>
				<key>line</key>
				<integer>120</integer>
			</dict>
		</dict>
		<key>tests/test-1m.out</key>
		<dict>
			<key>caret</key>
			<dict>
				<key>column</key>
				<integer>0</integer>
				<key>line</key>
				<integer>0</integer>
			</dict>
			<key>firstVisibleColumn</key>
			<integer>0</integer>
			<key>firstVisibleLine</key>
			<integer>0</integer>
		</dict>
		<key>tests/test-1m.py</key>
		<dict>
			<key>caret</key>
			<dict>
				<key>column</key>
				<integer>26</integer>
				<key>line</key>
				<integer>93</integer>
			</dict>
			<key>columnSelection</key>
			<false/>
			<key>firstVisibleColumn</key>
			<integer>0</integer>
			<key>firstVisibleLine</key>
			<integer>48</integer>
			<key>selectFrom</key>
			<dict>
				<key>column</key>
				<integer>22</integer>
				<key>line</key>
				<integer>93</integer>
			</dict>
			<key>selectTo</key>
			<dict>
				<key>column</key>
				<integer>26</integer>
				<key>line</key>
				<integer>93</integer>
			</dict>
		</dict>
		<key>tests/test-1s.out</key>
		<dict>
			<key>caret</key>
			<dict>
				<key>column</key>
				<integer>30</integer>
				<key>line</key>
				<integer>45</integer>
			</dict>
			<key>firstVisibleColumn</key>
			<integer>0</integer>
			<key>firstVisibleLine</key>
			<integer>15</integer>
		</dict>
		<key>tests/test-1s.py</key>
		<dict>
			<key>caret</key>
			<dict>
				<key>column</key>
				<integer>26</integer>
				<key>line</key>
				<integer>77</integer>
			</dict>
			<key>columnSelection</key>
			<false/>
			<key>firstVisibleColumn</key>
			<integer>0</integer>
			<key>firstVisibleLine</key>
			<integer>32</integer>
			<key>selectFrom</key>
			<dict>
				<key>column</key>
				<integer>22</integer>
				<key>line</key>
				<integer>77</integer>
			</dict>
			<key>selectTo</key>
			<dict>
				<key>column</key>
				<integer>26</integer>
				<key>line</key>
				<integer>77</integer>
			</dict>
		</dict>
		<key>tests/test-1t.out</key>
		<dict>
			<key>caret</key>
			<dict>
				<key>column</key>
				<integer>0</integer>
				<key>line</key>
				<integer>0</integer>
			</dict>
			<key>firstVisibleColumn</key>
			<integer>0</integer>
			<key>firstVisibleLine</key>
			<integer>54</integer>
		</dict>
		<key>tests/test-1t.py</key>
		<dict>
			<key>caret</key>
			<dict>
				<key>column</key>
				<integer>26</integer>
				<key>line</key>
				<integer>103</integer>
			</dict>
			<key>columnSelection</key>
			<false/>
			<key>firstVisibleColumn</key>
			<integer>0</integer>
			<key>firstVisibleLine</key>
			<integer>58</integer>
			<key>selectFrom</key>
			<dict>
				<key>column</key>
				<integer>22</integer>
				<key>line</key>
				<integer>103</integer>
			</dict>
			<key>selectTo</key>
			<dict>
				<key>column</key>
				<integer>26</integer>
				<key>line</key>
				<integer>103</integer>
			</dict>
		</dict>
		<key>tests/test-ups.out</key>
		<dict>
			<key>caret</key>
			<dict>
				<key>column</key>
				<integer>42</integer>
				<key>line</key>
				<integer>22</integer>
			</dict>
			<key>columnSelection</key>
			<false/>
			<key>firstVisibleColumn</key>
			<integer>0</integer>
			<key>firstVisibleLine</key>
			<integer>0</integer>
			<key>selectFrom</key>
			<dict>
				<key>column</key>
				<integer>0</integer>
				<key>line</key>
				<integer>9</integer>
			</dict>
			<key>selectTo</key>
			<dict>
				<key>column</key>
				<integer>0</integer>
				<key>line</key>
				<integer>23</integer>
			</dict>
		</dict>
		<key>tests/test-ups.py</key>
		<dict>
			<key>caret</key>
			<dict>
				<key>column</key>
				<integer>32</integer>
				<key>line</key>
				<integer>102</integer>
			</dict>
			<key>columnSelection</key>
			<false/>
			<key>firstVisibleColumn</key>
			<integer>0</integer>
			<key>firstVisibleLine</key>
			<integer>57</integer>
			<key>selectFrom</key>
			<dict>
				<key>column</key>
				<integer>28</integer>
				<key>line</key>
				<integer>102</integer>
			</dict>
			<key>selectTo</key>
			<dict>
				<key>column</key>
				<integer>32</integer>
				<key>line</key>
				<integer>102</integer>
			</dict>
		</dict>
	</dict>
	<key>openDocuments</key>
	<array>
		<string>makefile</string>
		<string>COPYING</string>
		<string>tests/test-ups.py</string>
		<string>tests/test-ups.out</string>
		<string>tests/test-1.py</string>
		<string>tests/coarse_3pt_etas_Ds.bint8</string>
		<string>tests/test-1b.py</string>
		<string>tests/test-1t.out</string>
		<string>tests/test-1b.out</string>
		<string>tests/test-1m.py</string>
		<string>tests/test-1s.py</string>
		<string>tests/test-1s.out</string>
		<string>tests/coarse_3pt_1link_Ds_and_etas_kinCth1.10.576.bint8</string>
		<string>tests/test-1t.py</string>
		<string>tests/makefile</string>
		<string>corrfitter.py</string>
		<string>dataset.py</string>
		<string>tests/test-1m.out</string>
		<string>CHANGES</string>
		<string>setup.py</string>
		<string>README</string>
		<string>doc/source/index.rst</string>
		<string>doc/source/Makefile</string>
		<string>doc/source/conf.py</string>
		<string>doc/source/corrfitter.rst</string>
		<string>tests/README</string>
		<string>MANIFEST.in</string>
		<string>INSTALLATION</string>
	</array>
	<key>showFileHierarchyDrawer</key>
	<true/>
	<key>windowFrame</key>
	<string>{{191, 19}, {561, 774}}</string>
</dict>
</plist>
