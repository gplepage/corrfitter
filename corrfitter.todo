- Add a median() function (like in lsqfit.Accumulator) to Dataset.

- Fix assemble(): doesn't work now if template contains reg-expression symbols (like +) since the template elements are used as regular expressions in one place (grep()) but later as literal strings. Must be all the same.