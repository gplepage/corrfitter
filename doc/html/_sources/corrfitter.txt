:mod:`corrfitter` - Least-Squares Fit to Correlators
=====================================================

.. moduleauthor:: G.P. Lepage <g.p.lepage@cornell.edu>

.. |CorrFitter| replace:: :class:`corrfitter.CorrFitter`
.. |Corr2| replace:: :class:`corrfitter.Corr2`
.. |Corr3| replace:: :class:`corrfitter.Corr3`
.. |BaseModel| replace:: :class:`corrfitter.BaseModel`
.. |Dataset| replace:: :class:`gvar.dataset.Dataset`
.. |GVar| replace:: :class:`gvar.GVar`

.. automodule:: corrfitter
   :synopsis: Least-squares fitter for correlators.


Correlator Model Objects
-------------------------
Correlator objects describe theoretical models that are fit to
correlator data by varying the models' parameters. 

A model object's parameters are specified through priors for the fit. A
model assigns labels to each of its parameters (or arrays of related
parameters), and these labels are used to identify the corresponding
parameters in the prior. Parameters can be shared by more than one model
object.

A model object also specifies the data that it is to model. The data is 
identified by the data tag that labels it in the input file or |Dataset|.

.. autoclass:: corrfitter.Corr2
   :members:
   
.. autoclass:: corrfitter.Corr3
   :members:
   
.. autoclass:: corrfitter.BaseModel
   :members:


|CorrFitter| Objects
---------------------
|CorrFitter| objects are wrappers for :func:`lsqfit.nonlinear_fit()` which
is used to fit a collection of models to a collection of Monte Carlo data.

.. autoclass:: corrfitter.CorrFitter
   :members:

