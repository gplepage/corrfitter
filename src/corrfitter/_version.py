__version__ = '8.2'