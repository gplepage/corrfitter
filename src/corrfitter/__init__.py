"""
This module contains tools that facilitate least-squares fits, as functions
of time ``t``, of simulation (or other statistical) data for 2-point and
3-point correlators of the form::

    Gab(t)    =  <b(t) a(0)>
    Gavb(t,T) =  <b(T) V(t) a(0)>

where ``T > t > 0``. Each correlator is modeled using |Corr2| for 2-point
correlators, or |Corr3| for 3-point correlators in terms of amplitudes for
each source ``a``, sink ``b``, and vertex ``V``, and the energies
associated with each intermediate state. The amplitudes and energies are
adjusted in the least-squares fit to reproduce the data; they are defined
in a shared prior (typically a dictionary).

An object of type |CorrFitter| describes a collection of correlators and is
used to fit multiple models to data simultaneously. Fitting multiple
correlators simultaneously is important if there are statistical
correlations between the correlators. Any number of correlators may be
described and fit by a single |CorrFitter| object.

Typical code has the structure ::

    from corrfitter import CorrFitter

    data = make_data('mcfile')          # user-supplied routine
    models = make_models()              # user-supplied routine
    N = 4                               # number of terms in fit functions
    prior = make_prior(N)               # user-supplied routine
    fitter = CorrFitter(models=models)
    fit = fitter.lsqfit(data=data, prior=prior)  # do the fit
    print_results(fit, prior, data)     # user-supplied routine

where ``make_data`` assembles the correlator data to be fit,
``make_prior`` defines Bayesian priors for the fit parameters,
``fitter.lsqfit(...)`` does the fit, and ``print_results``
writes out the rsults. Sample code for each of these routines is
given in the tutorial documentation for |CorrFitter|.
"""

# Created by G. Peter Lepage, Cornell University, on 2010-11-26.
# Copyright (c) 2010-2021 G. Peter Lepage.
#
# This program is free software: you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation, either version 3 of the License, or
# any later version (see <http://www.gnu.org/licenses/>).
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.

from ._corrfitter import * 
from ._version import __version__