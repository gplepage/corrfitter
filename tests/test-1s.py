#!/usr/bin/env python
# encoding: utf-8
"""
test-1.py

Created by Peter Lepage on 2010-11-26.
Copyright (c) 2010/2011 Cornell University. All rights reserved.
"""

import os
from corrfitter import Corr2,Corr3,CorrFitter
from gvar import gvar,log,exp,evalcov,Dataset,BufferDict,avg_data
from numpy import array,arange,dot

import lsqfit

lsqfit.nonlinear_fit.fmt_parameter = '%7.3f +- %7.3f'

DISPLAYPLOTS = False         # display plots at end of fitting
try: 
    import matplotlib
except ImportError:
    DISPLAYPLOTS = False

TEST = True     # testing mode?

def main():
    # dfile = "coarse_3pt_etas_etas_kinCth1.10.576.bint8"  # data file
    dfile = "coarse_3pt_1link_Ds_and_etas_kinCth1.10.576.bint8"
    data = avg_data(Dataset(dfile,keys=['DsDsT18','DsDsT15','Ds']))
    fitter = CorrFitter(models=build_models())
    pfile = "test-1s.p" # last fit
    nexp_list = NEXP_TEST if TEST else [2,3,4,5,6,7,8]
    p0 = P0_TEST if TEST else pfile
    for nexp in nexp_list:
        print '========================== nexp =',nexp
        prior = build_prior(nexp)
        fit = fitter.lsqfit(data=data,prior=prior,p0=p0)
        print_results(fit,prior,data)
        print '\n\n'
    if DISPLAYPLOTS:
        fitter.display_plots()
##
      
def print_results(fit,prior,data):
    """ print out additional results from the fit """
    dE = exp(fit.p['logdE'])
    print 'dE =',fmtlist(dE[:3])
    print ' E =',fmtlist([sum(dE[:i+1]) for i in range(3)])
    V = fit.p['Vnn'][0]
    print 'm->V->m =',V.fmt()
    print
    outputs = {'V':V}
    inputs = {'stat':[data[k] for k in data]} # all data statistical errors
    inputs.update(prior)                      # all parts of the prior
    print fit.fmt_partialsdev(outputs,inputs)
##

def fmtlist(x):
    return '  '.join([xi.fmt() for xi in x])
##
    
def build_prior(nexp):
    """ build prior """
    prior = BufferDict()
    ## etas ##
    prior.add('a',[gvar(0.01,1.0) for i in range(nexp)])
    prior.add('logdE',[log(gvar(0.5,0.5)) for i in range(nexp)])
    prior['logdE'][0] = log(gvar(1.40,0.2))
    prior.add('ao',[gvar(0.01,1.0) for i in range(nexp)])
    prior.add('logdEo',[log(gvar(0.5,0.5)) for i in range(nexp)])
    prior['logdEo'][0] = log(gvar(1.59,0.2))
    ##
    ## V ##
    nV = (nexp*(nexp+1))/2
    prior.add('Vnn',[gvar(0.01,1.0) for i in range(nV)])
    prior.add('Voo',[gvar(0.01,1.0) for i in range(nV)])
    prior.add('Vno',[[gvar(0.01,1.0) for i in range(nexp)] for j in range(nexp)])
    return prior
##

def build_models():
    """ build models """
    tmin = 2
    tdata = range(64)
    tfit = range(tmin,64-tmin) # all ts
    
    tp = 64 # periodic
    models = [
        Corr2(datatag='Ds',tp=tp,tdata=tdata,tfit=tfit,
            a=('a','ao'),b=('a','ao'),dE=('logdE','logdEo'),s=(1.,-1.)),
            
        Corr3(datatag='DsDsT18',tdata=range(19),T=18,tfit=range(tmin,18-tmin),
            a=('a','ao'),dEa=('logdE','logdEo'),tpa=tp,sa=(1.,-1),
            b=('a','ao'),dEb=('logdE','logdEo'),tpb=tp,sb=(1.,-1.),
            Vnn='Vnn',Voo='Voo',Vno='Vno',Von='Vno',symmetric_V=True),
        #
        Corr3(datatag='DsDsT15',tdata=range(16),T=15,tfit=range(tmin,15-tmin),
            a=('a','ao'),dEa=('logdE','logdEo'),tpa=tp,sa=(1.,-1),
            b=('a','ao'),dEb=('logdE','logdEo'),tpb=tp,sb=(1.,-1.),
            Vnn='Vnn',Voo='Voo',Vno='Vno',Von='Vno',symmetric_V=True)
                    
    ][:]
    return models
##    

NEXP_TEST = [3]
P0_TEST = {'a': [0.21003188674777384, 0.20199445922587544,
0.5183807812003373], 'logdE': [0.27069323026592135, -0.9437824237909911,
-0.748844103009219], 'Vno': [[-0.21021700308173005, -0.008004198284218224,
0.0033815712672157078], [0.038037979385456035, -0.11929504858159,
0.011472143959295505], [0.002271934466225292, 0.1040458783723354,
0.009066523607814674]], 'Vnn': [0.10581237371666792, 0.0032368011577309395,
-0.027413371793672342, 0.015869286542962588, 0.027265956013929192,
0.010042601420795485], 'Voo': [-0.10552227151792866, 0.05790719018261943,
-0.08141353578942945, 0.003930130259594829, 0.01006758531249384,
0.009981448331701769], 'ao': [0.06116739717004869, 0.10794830765204166,
0.08915150248655652], 'logdEo': [0.33845814370750404, -1.366437774186797,
-1.0541868450784229]}

for k in P0_TEST:
    P0_TEST[k] = array(P0_TEST[k])

if __name__ == '__main__':
    main()

