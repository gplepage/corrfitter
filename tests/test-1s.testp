{"s:Vno": [[-0.21021700308173005, -0.008004198284218224, 0.0033815712672157078], [0.038037979385456035, -0.11929504858159, 0.011472143959295505], [0.002271934466225292, 0.1040458783723354, 0.009066523607814674]], "s:logdEo": [0.33845814370750404, -1.366437774186797, -1.0541868450784229], "keys": ["s:a", "s:logdE", "s:ao", "s:logdEo", "s:Vnn", "s:Voo", "s:Vno"], "s:a": [0.21003188674777384, 0.20199445922587544, 0.5183807812003373], "s:logdE": [0.27069323026592135, -0.9437824237909911, -0.748844103009219], "s:Vnn": [0.10581237371666792, 0.0032368011577309395, -0.027413371793672342, 0.015869286542962588, 0.027265956013929192, 0.010042601420795485], "s:ao": [0.06116739717004869, 0.10794830765204166, 0.08915150248655652], "s:Voo": [-0.10552227151792866, 0.05790719018261943, -0.08141353578942945, 0.003930130259594829, 0.01006758531249384, 0.009981448331701769]}