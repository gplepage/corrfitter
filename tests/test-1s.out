========================== nexp = 3
Least Square Fit:
  chi2/dof [dof] = 0.45 [56]    Q = 1    logGBF = -145.91    itns = 1

Parameters:
              a_  0.210 +-   0.001           (    0.01 +-        1)
               _  0.202 +-   0.024           (    0.01 +-        1)
               _  0.518 +-   0.005           (    0.01 +-        1)
          logdE_  0.271 +-   0.000           (    0.34 +-     0.14)
               _ -0.944 +-   0.101           (   -0.69 +-        1)
               _ -0.749 +-   0.037           (   -0.69 +-        1)
             ao_  0.061 +-   0.009           (    0.01 +-        1)
               _  0.108 +-   0.032           (    0.01 +-        1)
               _  0.089 +-   0.035           (    0.01 +-        1)
         logdEo_  0.338 +-   0.011           (    0.46 +-     0.13)
               _ -1.366 +-   0.355           (   -0.69 +-        1)
               _ -1.054 +-   0.776           (   -0.69 +-        1)
            Vnn_  0.106 +-   0.002           (    0.01 +-        1)
               _  0.003 +-   0.005           (    0.01 +-        1)
               _ -0.027 +-   0.004           (    0.01 +-        1)
               _  0.016 +-   0.999           (    0.01 +-        1)
               _  0.027 +-   0.997           (    0.01 +-        1)
               _  0.010 +-   1.000           (    0.01 +-        1)
            Voo_ -0.106 +-   0.098           (    0.01 +-        1)
               _  0.058 +-   0.128           (    0.01 +-        1)
               _ -0.081 +-   0.191           (    0.01 +-        1)
               _  0.004 +-   0.999           (    0.01 +-        1)
               _  0.010 +-   1.000           (    0.01 +-        1)
               _  0.010 +-   1.000           (    0.01 +-        1)
            Vno_ -0.210 +-   0.009           (    0.01 +-        1)
               _ -0.008 +-   0.042           (    0.01 +-        1)
               _  0.003 +-   0.039           (    0.01 +-        1)
               _  0.038 +-   0.081           (    0.01 +-        1)
               _ -0.119 +-   0.988           (    0.01 +-        1)
               _  0.011 +-   1.000           (    0.01 +-        1)
               _  0.002 +-   0.054           (    0.01 +-        1)
               _  0.104 +-   0.993           (    0.01 +-        1)
               _  0.009 +-   1.000           (    0.01 +-        1)

dE = 1.3109(7)  0.39(4)  0.47(2)
 E = 1.3109(7)  1.70(4)  2.17(2)
m->V->m = 0.106(2)

Partial % Errors:
                             V
------------------------------
                  a:      0.00
               stat:      0.48
                Vno:      0.02
                Vnn:      1.55
                Voo:      0.33
              logdE:      0.02
                 ao:      0.01
             logdEo:      0.09
------------------------------
              total:      1.66




